pkg/resources/server.go

```diff
@@ -115,6 +117,19 @@ func (rs *resourceServer) NotifyRegistrationStatus(ctx context.Context,
 	return &registerapi.RegistrationStatusResponse{}, nil
 }
 
+func (rs *resourceServer) GetPreferredAllocation(ctx context.Context,
+	rqt *pluginapi.PreferredAllocationRequest) (*pluginapi.PreferredAllocationResponse, error) {
+	glog.Infof("GetPreferredAllocation called with %+v", rqt)
+	resp := new(pluginapi.PreferredAllocationResponse)
+	for _, containerReq := range rqt.ContainerRequests {
+		containerResp := new(pluginapi.ContainerPreferredAllocationResponse)
+		containerResp.DeviceIDs = rs.allocator.Allocate(containerReq, rs.resourcePool)
+		resp.ContainerResponses = append(resp.ContainerResponses, containerResp)
```

Comment:
We need to use default allocator or check for rs.allocator == nil`

Reply:
added nil check.